### PR TITLE
Update permissions documentation

### PR DESCRIPTION
**Pull request by simongroenewolt**
I'll update the old 3.7 docs for the permissions config and setup.

### CODE REVIEW

**Comment by I-Valchev** [comment]
Hey @simongroenewolt, is this still on the horizon? We're getting nearer the official Bolt 5 release, and the powerful permissions feature will definitely be one of the center pieces of the release :-)

**Comment by simongroenewolt** [comment]
@I-Valchev @bobdenotter I finally wrote something / reviewed and adapted the old docs. Maybe if one of you reads it and asks a couple of questions I can do one extra update and we'll call it 'goed genoeg'?

File: docs/configuration/permissions.md [hunk 1]
```diff
@@ -8,10 +8,10 @@ The Bolt Permissions System
 Bolt uses a [Role-Based Permission](https://en.wikipedia.org/wiki/Role-based_access_control)
 system. This means that:
 
-* Every *user* can be a member of zero or more *roles*
+* Every *user* can have zero or more *roles*
```
Comment (bobdenotter) [inline docs/configuration/permissions.md]: I think "can have" can be replaced with "has". It's not optional to have 'zero or more roles'
```suggestion
* Every *user* has zero or more *roles*
```

File: docs/configuration/permissions.md [hunk 2]
```diff
@@ -8,10 +8,10 @@ The Bolt Permissions System
...
 * Every *role* can grant zero or more *permissions*
 * The same *permission* can be granted through several different *roles*
-* Several users can be members of the same *role*
+* Several users can have the same *role*
```
Comment (bobdenotter) [inline docs/configuration/permissions.md]:
```suggestion
* Several users can have the same *role*, and one role can be granted to multiple *users*
```

File: docs/configuration/permissions.md [hunk 3]
```diff
@@ -32,14 +32,17 @@ permissions you don't want them to have. Two important considerations:
...
-    `root`, or take away `root` from others. In short: **Do _not_ give someone
-    `edit users` permission, unless you trust them fully!!**
+ 2. If somebody has the permission to `user:edit`, they can also grant
+    permissions to themselves or others. This means they can give themselves
+    roles with more permissions, or take away those roles from others.
+    This is similar for a lot of permissions that are by default assigned
+    to ROLE_ADMIN only.
+    In short: **Do _not_ give someone the `ROLE_ADMIN` (or `ROLE_DEVELOPER`) role, 
+    or change the setup in a way that they have permission to do things that were
+    previously only accessible to ROLE_ADMIN unless you trust them fully!!**
```
Comment (bobdenotter) [inline docs/configuration/permissions.md]:
```suggestion
    previously only accessible to `ROLE_ADMIN` unless you trust them fully!**
```

File: docs/configuration/permissions.md [hunk 4]
```diff
@@ -169,77 +180,62 @@ content, but only the chief editor can decide if and when it is published. Each
...
+Start by reading the linked documentation when you want to make a change.
+The Bolt classes that implement the Bolt specific functionality are 
+in the `Bolt\Security` namespace. `ContentVoter` and `GlobalVoter` are doing most
+of the decision making. (Strictly speaking they are only 'voting' and the decision is
+made elsewhere in the security system)
 
-```yaml
-debug_permission_audit_mode: true
-```
+If you search the bolt core code for `isGranted` and the templates for `is_granted` you will
```
Comment (bobdenotter) [inline docs/configuration/permissions.md]:
```suggestion
If you search the Bolt core code for `isGranted` and the templates for `is_granted` you will 
```

File: docs/configuration/permissions.md [hunk 5]
```diff
@@ -32,14 +32,17 @@ permissions you don't want them to have. Two important considerations:
 
  1. Permissions are quite central to Bolt's inner workings, and by
     misconfiguring them, you can lock yourself out - for example, removing the
-    `anonymous` role from the `login` or `postLogin` permissions will make
-    logging in completely impossible: you will not even be allowed to see the
-    login page.
+    `IS_AUTHENTICATED_REMEMBERED` role from the `dashboard` permission will make
+    the bolt dashboard completely inaccessible for everyone.
```
Comment (bobdenotter) [inline docs/configuration/permissions.md]:
```suggestion
    the Bolt dashboard completely inaccessible for everyone.
```

File: docs/configuration/permissions.md [hunk 6]
```diff
@@ -127,39 +128,49 @@ default`; but if the permission is not mentioned, the corresponding entry in
...
+These roles are put in a hierarchy specified in `config/packages/security.yaml`, this 
+means that in the order specified above every role also has the permissions given to
+the roles that are following it. (So `ROLE_ADMIN` can do everything `ROLE_CHIEF_EDITOR` can do etc.)
+
+Below is an explanation of permissions for these roles in general terms, as
+configured by default. Always check `config/bolt/permissions.yaml` if you want
+to be certain.
+
+`ROLE_USER` is the role a user gets when there is no role set. It doesn't
+give access to anything in the bolt admin.
```
Comment (bobdenotter) [inline docs/configuration/permissions.md]:
```suggestion
give access to anything in the Bolt administration panel.
```

**Review by bobdenotter — APPROVED** [review]
Hi Simon! Glad that you've made these improvements. 

I totally agree, let's tweak a few minor things, and let's roll with this! 👍 

(I've changed the target for this PR to `5.0`, but that hopefully doesn't affect your local checkout)

**Comment by simongroenewolt** [comment]
I've applied all suggestions.

**Comment by bobdenotter** [comment]
Allrighty! Then I'mma merge it in now, and we'll undoubtedly do some more fine-tuning in the coming weeks! 👍

**Comment by I-Valchev** [comment]
Great to have improved permissions docs, thanks @simongroenewolt ! 👍

**Pull request merged**